src/boxed.rs

```diff
@@ -180,15 +179,14 @@ macro_rules! glib_boxed_wrapper {
                 res
             }
 
-            unsafe fn from_glib_full_num_as_vec(mut ptr: *mut *mut $ffi_name, num: usize) -> Vec<Self> {
+            unsafe fn from_glib_full_num_as_vec(ptr: *mut *mut $ffi_name, num: usize) -> Vec<Self> {
                 if num == 0 || ptr.is_null() {
                     return Vec::new();
                 }
 
                 let mut res = Vec::with_capacity(num);
-                for _ in 0..num {
-                    res.push($crate::translate::from_glib_full(ptr::read(ptr)));
-                    ptr = ptr.offset(1);
+                for i in 0..num {
+                    res.push($crate::translate::from_glib_full(ptr::read(ptr.offset(i as isize))));
                 }
                 glib_ffi::g_free(ptr as *mut _);
```

Comment:
The point here is to also remove the mut variable to prevent such sillyness in the future. It's equivalent code-wise apart from that.